src/sentry/api/event_search.py

```diff
@@ -783,9 +786,15 @@ def to_list(value):
     for term in parsed_terms:
         if isinstance(term, SearchFilter):
             name = term.key.name
-            if name == PROJECT_KEY:
+            if name in (PROJECT_NAME_ALIAS, PROJECT_ALIAS):
                 if projects is None:
                     projects = get_projects(params)
+                project = projects.get(term.value.value)
+                if not project:
+                    raise InvalidSearchQuery(
+                        "Invalid query. Project %s must exist and be in global header"
```

Comment:
```suggestion
                        "Invalid query. Project %s does not exist or is not an actively selected project."
```
Not sure if this is better :golf:

Reply:
It's better.